### DISCUSSION TITLE
Group by + Select

### DISCUSSION THREAD

**Discussion by MarvelousWololo** [prisma/prisma#19014] [Q&A]
Hello there, everyone! Is there anyway to do something similar to this? https://debuxing.com/entity-framework-group-by-example-c

```
using (RobotIndustryEntities context = [new](http://www.google.com/search?q=new+msdn.microsoft.com) RobotIndustryEntities())
{
    var robotFactories = context
        .RobotFactories.GroupBy(x => x.Location)
        .Select(x => [new](http://www.google.com/search?q=new+msdn.microsoft.com) { Location = x.Key, Buildings = x.Count() });
}
```

**Comment by nurul3101** [comment]
Hello @MarvelousWololo 👋 

It looks like you want to group records by a specific field and then perform aggregations.
Can you have a look at this [example](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#examples-16)?

Your query could look something similar to this:
```ts
  const robotFactories = await prisma.robotFactory.groupBy({
    by: ['location'],
    _count: {
      location: true,
    },
  });
```

**Reply by nurul3101** [reply]
Let us know in case this helps or if you need any assistance.

**Reply by nurul3101** [reply]
@MarvelousWololo If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
